src/SDL_internal.h

```diff
@@ -185,6 +185,46 @@
 
 #include <SDL3/SDL.h>
 #include <SDL3/SDL_intrin.h>
+
+
+#ifdef __ARM_NEON
```

Comment:
Do we want SDL_DISABLE_NEON as an option in the CMake config?

Reply:
Looking at it now.
https://github.com/libsdl-org/SDL/blob/f18b5656f6f859e4d4e096d290afd9fae884a5b8/CMakeLists.txt#L372-L373
ARM simd and neon is currently only allowed for 32 bit arm.
I suppose these are also supported on 64-bit arm?
They are also disabled by default. Perhaps this is a good opportunity to enable them by default?

Reply:
The PlayStation Vita CMake code has this piece of commented out code:
https://github.com/libsdl-org/SDL/blob/f18b5656f6f859e4d4e096d290afd9fae884a5b8/CMakeLists.txt#L2523-L2534
This does not enable arm simd at all. It only affects CMake stdout reporting.

Reply:
I fixed ARMNEON and ARMSIMD detection in CMake.
I also see why it is disabled by default (other license as SDL).

It fails to assemble with the android assembler.
I think it's because it uses directives which llvm does not recognize.

error log: [android-assemble-pixman-arm-neon-asm-S.txt](https://github.com/libsdl-org/SDL/files/10832950/android-assemble-pixman-arm-neon-asm-S.txt)

Reply:
I tried building SDL for vita's gcc toolchain (vita uses 32-bit ARM). The file could be assembled successfully there

However, linking the tests fail with:
```
[1/67] Linking C executable test/testoffscreen
ninja: job failed: : && /home/user/vitasdk/bin/arm-vita-eabi-gcc -Wl,-q -Wl,-z,nocopyreloc test/CMakeFiles/testoffscreen.dir/testoffscreen.c.obj -o test/testoffscreen  libSDL3_test.a  libSDL3.a  -lSceGxm_stub  -lSceDisplay_stub  -lSceCtrl_stub  -lSceAppMgr_stub  -lSceAppUtil_stub  -lSceAudio_stub  -lSceAudioIn_stub  -lSceSysmodule_stub  -lSceIofilemgr_stub  -lSceCommonDialog_stub  -lSceTouch_stub  -lSceHid_stub  -lSceMotion_stub  -lScePower_stub  -lSceProcessmgr_stub  -lm && :
/home/user/vitasdk/bin/../lib/gcc/arm-vita-eabi/10.3.0/../../../../arm-vita-eabi/bin/ld: libSDL3.a(SDL_fillrect.c.obj): in function `fill_8_neon':
SDL_fillrect.c:(.text+0x2b2): undefined reference to `FillSurfaceRect8ARMNEONAsm'
/home/user/vitasdk/bin/../lib/gcc/arm-vita-eabi/10.3.0/../../../../arm-vita-eabi/bin/ld: libSDL3.a(SDL_fillrect.c.obj): in function `fill_16_neon':
SDL_fillrect.c:(.text+0x2de): undefined reference to `FillSurfaceRect16ARMNEONAsm'
/home/user/vitasdk/bin/../lib/gcc/arm-vita-eabi/10.3.0/../../../../arm-vita-eabi/bin/ld: libSDL3.a(SDL_fillrect.c.obj): in function `fill_32_neon':
SDL_fillrect.c:(.text+0x308): undefined reference to `FillSurfaceRect32ARMNEONAsm'
/home/user/vitasdk/bin/../lib/gcc/arm-vita-eabi/10.3.0/../../../../arm-vita-eabi/bin/ld: libSDL3.a(SDL_fillrect.c.obj): in function `fill_8_simd':
SDL_fillrect.c:(.text+0x330): undefined reference to `FillSurfaceRect8ARMSIMDAsm'
/home/user/vitasdk/bin/../lib/gcc/arm-vita-eabi/10.3.0/../../../../arm-vita-eabi/bin/ld: libSDL3.a(SDL_fillrect.c.obj): in function `fill_16_simd':
SDL_fillrect.c:(.text+0x35c): undefined reference to `FillSurfaceRect16ARMSIMDAsm'
/home/user/vitasdk/bin/../lib/gcc/arm-vita-eabi/10.3.0/../../../../arm-vita-eabi/bin/ld: libSDL3.a(SDL_fillrect.c.obj): in function `fill_32_simd':
SDL_fillrect.c:(.text+0x386): undefined reference to `FillSurfaceRect32ARMSIMDAsm'
collect2: error: ld returned 1 exit status
ninja: subcommand failed
```
I cannot find implementations of these symbols.

Reply:
Fixed in https://github.com/libsdl-org/SDL/pull/7368/commits/532257fcf053d3be4b86fba6f589ee96fd982969, I think (please review)

Reply:
Yup, that looks correct, thanks!